react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java

```diff
@@ -81,6 +84,28 @@ public static MediaType getEnum(String value) {
         }
     }
 
+    enum GutenbergUserEvent {
+        PAGE_TEMPLATE_APPLIED("page_template_applied");
+
+        private static final Map<String, GutenbergUserEvent> MAP = new HashMap<>();
+
+        static {
+            for (GutenbergUserEvent event : values()) {
+                MAP.put(event.name, event);
+            }
+        }
+
+        String name;
+
+        GutenbergUserEvent(String name) {
+            this.name = name;
+        }
+
+        public static GutenbergUserEvent getEnum(String eventName) {
+            return MAP.get(eventName);
+        }
+    }
```

Comment:
Do you think it would work to simplify this enum a bit along these lines?

```
enum GutenbergUserEvent { 
  PAGE_TEMPLATE_APPLIED;

  @Nullable 
  public static getEnum(String eventName) {
    switch (eventName) {
      case "page_template_applied": return PAGE_TEMPLATE_APPLIED;
      default: null;
    }
  }
}
```

Reply:
The benefit of using the static map and static initializer is that we only need to add new events in one place. Your example is cleaner, but will require new entries in two places. Not a huge issue, but that's the reason I went with the static map.